hw4-bombard/bombard.js

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env node
+const yargs = require('yargs/yargs');
+const { hideBin } = require('yargs/helpers');
+const { table } = require('table');
+const axios = require('axios');
+
+const argv = yargs(hideBin(process.argv))
+  .command('$0 <url>', 'server benchmarking tool', yargs => {
+    yargs
+      .positional('url', {
+        describe: 'url to send requests to',
+        type: 'string',
+      })
+      .option('c', {
+        alias: 'concurrency',
+        describe: 'number of parallel requests to perform at a time',
+        type: 'number',
+        default: 1,
+      })
+      .option('n', {
+        alias: 'requests',
+        describe: 'number of requests to perform for the benchmarking session',
+        type: 'number',
+        default: 1,
+      })
+      .option('b', {
+        alias: 'body',
+        describe: 'send a random generated body with request',
+        type: 'boolean',
+        default: false,
+      });
+  })
+  .help()
+  .parse();
+
+const bombard = async ({ url, concurrency, requests, body }) => {
+  const results = [];
+  const promises = [];
+
+  const request = async () => {
+    while (true) {
+      const currentRequestNumber = results.length;
+      if (currentRequestNumber >= requests) break;
+
+      const start = Date.now();
+      try {
+        const requestData = body ? { data: Math.random().toString(36).substring(2) } : {};
+        await axios.post(url, requestData);
+        const time = Date.now() - start;
+
+        results[currentRequestNumber] = time;
+      } catch (error) {
+        console.error('Request failed:', error.message);
+      }
+    }
+  };
+
+  for (let i = 0; i < concurrency; i++) {
```

Comment:
if concurrency is 1 and requests number is 10 than bombard will call web server 1 times in total